### PR TITLE
feat(conformance): Enable experiment API tests in KF mode.

### PR DESCRIPTION
**Pull request by james-jwu**
Added 3 flags:
- isDebugMode is enables HTTP request/response logging
- IsKubeflowMode enables the tests in full Kubeflow mode
- resourceNamespace: the namespace/profile under which the test
  resources are created

Added a new HTTP client that uses SA token volume projection auth. The
test pods will be set up to project SA token.

Plumbed everything through for experiment API tests. The other tests
will be enabled in subsequent PRs.

### CODE REVIEW

File: backend/src/common/client/api_server/util.go [hunk 1]
```diff
@@ -17,14 +21,33 @@ import (
...
+var SATokenVolumeProjectionAuth runtime.ClientAuthInfoWriter = runtime.ClientAuthInfoWriterFunc(
+	func(r runtime.ClientRequest, _ strfmt.Registry) error {
+		var projectedPath string
+		if projectedPath = os.Getenv(saTokenPathEnvVar); projectedPath == "" {
+			projectedPath = saDefaultTokenPath
+		}
+
+		content, err := ioutil.ReadFile(projectedPath)
+		if err != nil {
+			log.Fatal(err)
```
Comment (chensun) [inline backend/src/common/client/api_server/util.go:44]: return error here?

**Review by chensun — COMMENTED** [review]
left a few nitpicks, overall LGTM.

Thanks!

/lgtm

File: backend/src/common/client/api_server/experiment_client.go [hunk 2]
```diff
@@ -36,9 +38,27 @@ func NewExperimentClient(clientConfig clientcmd.ClientConfig, debug bool) (
...
+		authInfoWriter: PassThroughAuth,
+	}, nil
+}
+
+func NewKubeflowInClusterExperimentClient(namespace string, debug bool) (
+	*ExperimentClient, error) {
+
+	runtime, err := NewKubeflowInClusterHTTPRuntime(namespace, debug)
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (james-jwu) [reply backend/src/common/client/api_server/experiment_client.go]: Done.

File: backend/src/common/client/api_server/job_client.go [hunk 3]
```diff
@@ -37,12 +39,29 @@ func NewJobClient(clientConfig clientcmd.ClientConfig, debug bool) (
...
 		apiClient: apiClient,
 	}, nil
 }
 
+func NewKubeflowInClusterJobClient(namespace string, debug bool) (
+	*JobClient, error) {
+
+	runtime, err := NewKubeflowInClusterHTTPRuntime(namespace, debug)
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (james-jwu) [reply backend/src/common/client/api_server/job_client.go]: Done. Also updated other clients.

File: backend/src/common/client/api_server/util.go (hunk 1 above)
Comment (james-jwu) [reply backend/src/common/client/api_server/util.go]: Done.

**Comment by chensun** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**